DM-7261: Port to Python 3

timj opened the pull request.

r-owen committed:
Run autopep8 on Python code

r-owen committed:
Update .gitignore

Add _build.* doc/xml/ and tests/.cache to .gitignore

r-owen committed:
Run futurize -1 -w .

r-owen committed:
Run futurize -2 -x division_safe -w .

Also remove one obsolete "from __future__ import"
that was confusing futurize

timj commented on .gitignore:
```diff
@@ -14,6 +15,7 @@ doc/html
 doc/*.tag
 doc/*.inc
 doc/doxygen.conf
+doc/xml
```
Also `tests/.cache` I think.

r-owen replied:
I should add that. Oddly enough when I run py.test I don't see this; I'm not sure why

timj commented on examples/makeNoiseCoadd.py:
```diff
@@ -1,30 +1,32 @@
+from builtins import range
 """Make a coadd from gaussian noise images
 """
+from __future__ import print_function
```
We usually put the `__future__` lines before other imports.

timj commented on examples/plotHistogram.py:
```diff
@@ -1,30 +1,32 @@
+from builtins import zip
 """Plot a histogram for a chi squared coadd and overlay a chi squared distribution
 """
+from __future__ import print_function
```
Same comment about grouping the `__future__` imports.

r-owen replied:
I took a manual pass through all imports.

timj commented on examples/plotHistogram.py:
```diff
@@ -1,30 +1,32 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
 from __future__ import with_statement
```
This can be removed. It's not relevant for 2.7.

r-owen replied:
Good catch. I found this in several spots.

r-owen committed:
Add python_future to required packages

r-owen committed:
Change "== None" to "is None"

r-owen committed:
Standardize imports

Remove obsolete "from __future__ import with_statement"
Put imports into standard order and grouping
Remove unused imports
Change "import numpy" to "import numpy as np"

r-owen committed:
Fix wrapping and indentation

Fixed a few lines that were too long or improperly indented
